### PR TITLE
fix_enable

### PR DESCRIPTION
**Pull request by bit-dance**
### What problem does this PR solve?
In multi mount situation,the xattr information will wrong and we can not use the sum to calculate the file bytes.
Issue Number: #2094 

Problem Summary:
When the fs mount is mounted multiple times with different enableSumInDir switches, the xattr information recorded in the metadata will be inaccurate.
### What is changed and how it works?
If mds find there are multi mounts and then it will close the enableSumInDir forever.
What's Changed:
fuse_client.cpp fs_info_wrapper.cpp
How it Works:
just like above
Side effects(Breaking backward compatibility? Performance regression?):
Maybe we can't use the sum xattr later if we close the enableSumInDir.
### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by zhanghuidinah** [comment]
cicheck

**Comment by bit-dance** [comment]
cicheck

**Comment by zhanghuidinah** [comment]
cicheck

**Comment by caoxianfei1** [comment]
image
curve_fuse is not found @bit-dance

**Comment by bit-dance** [comment]
Ok,i will research this part later

**Comment by SeanHai** [comment]
cicheck

File: curvefs/src/mds/fs_info_wrapper.cpp [hunk 1]
```diff
@@ -113,6 +113,9 @@ void FsInfoWrapper::AddMountPoint(const Mountpoint& mp) {
     *p = mp;
 
     fsInfo_.set_mountnum(fsInfo_.mountnum() + 1);
+    if (fsInfo_.mountnum() > 1) {
```
Comment (SeanHai) [inline curvefs/src/mds/fs_info_wrapper.cpp:116]: // stats are not cached in xattr in multi-mount scenarios
if (fsInfo_.enablesumindir() && fsInfo_.mountnum() > 1) {
    fsInfo_.set_enablesumindir(false);
}

**Comment by SeanHai** [comment]
@bit-dance 
The current implementation is incomplete. In the multi-mount scenario, the flag in fsInfo is set to false, and subsequent mount points perceive this change (enableSumInDir_ in the client will be a new value), but the mounted client The enableSumInDir_ in is still old, so it is necessary to reversely update the change of this value from mds. At present, the connection from clinet to mds is realized by session. You can consider taking back the flag from mds in the renewal response and updating it to the client.

**Comment by Cyber-SiKu** [comment]
@bit-dance  Will you continue to follow up this PR?

**Comment by bit-dance** [comment]
Yes,these times i am dealing with my exams and later i will continue following codes.

**Comment by caoxianfei1** [comment]
@bit-dance feel free to continue.

**Comment by bit-dance** [comment]
@caoxianfei1 I am so sorry but i will have time to continue next week.

**Pull request closed without merging**